tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -334,6 +334,367 @@ TILEDB_EXPORT int32_t tiledb_query_get_status_details(
+ * tiledb_group_open(ctx, group, TILEDB_READ);
+ * tiledb_group_close(ctx, group);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param group The group object to be closed.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note If the group object has already been closed, the function has
+ *     no effect.
+ */
+TILEDB_EXPORT int32_t
+tiledb_group_close(tiledb_ctx_t* ctx, tiledb_group_t* group);
+
+/**
+ * Creates a new TileDB group.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_group_t* group;
+ * tiledb_group_alloc(ctx, "my_group", &group);
+ * tiledb_group_free(&group);
+ * @endcode
+ *
+ * @param group The TileDB group to be freed
+ */
+TILEDB_EXPORT void tiledb_group_free(tiledb_group_t** group);
+
+/**
+ * Sets the group config.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_group_t* group;
+ * tiledb_group_alloc(ctx, "s3://tiledb_bucket/my_group", &group);
+ * tiledb_group_open(ctx, group, TILEDB_READ);
+ * // Set the config for the given group.
+ * tiledb_config_t* config;
+ * tiledb_group_set_config(ctx, group, config);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param group The group to set the config for.
+ * @param config The config to be set.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note The group does not need to be opened via `tiledb_group_open_at` to use
+ *      this function.
+ * @note The config should be set before opening an group.
+ */
+TILEDB_EXPORT int32_t tiledb_group_set_config(
+    tiledb_ctx_t* ctx, tiledb_group_t* group, tiledb_config_t* config);
+
+/**
+ * Gets the group config.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * // Retrieve the config for the given group.
+ * tiledb_config_t* config;
+ * tiledb_group_get_config(ctx, group, config);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param group The group to set the config for.
+ * @param config Set to the retrieved config.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_group_get_config(
+    tiledb_ctx_t* ctx, tiledb_group_t* group, tiledb_config_t** config);
+
+/**
+ * It puts a metadata key-value item to an open group. The group must
+ * be opened in WRITE mode, otherwise the function will error out.
+ *
+ * @param ctx The TileDB context.
+ * @param group An group opened in WRITE mode.
+ * @param key The key of the metadata item to be added. UTF-8 encodings
+ *     are acceptable.
+ * @param value_type The datatype of the value.
+ * @param value_num The value may consist of more than one items of the
+ *     same datatype. This argument indicates the number of items in the
+ *     value component of the metadata.
+ * @param value The metadata value in binary form.
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ *
+ * @note The writes will take effect only upon closing the group.
+ */
+TILEDB_EXPORT int32_t tiledb_group_put_metadata(
```

Comment:
The metadata documentation changed to remove the array-specific nature of it, yet here there's an API function with a group distinction.

No need to change this in this release, but next release the experimental API should unify metadata with a single set of calls for either array or group.